### PR TITLE
Adds the condenser, a new assembly component.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

This adds the twin brother to the igniter, the freezer. It does just about what it says on the tin, which is that it can be used to slowly cool down the air when used, just as an igniter can be used to slowly heat up the surrounding air when used. Could be used for improvised ice skating rinks, fire fighting equipment, or even just plain air conditioning when in a hotter environment than the station.
image

Also included is some basic cleanup of igniter code, since the heat variable I don't think was being used by igniters correctly, or in the slightest as far as I'm aware.

- [x] Add to component vendors as a premium item.

## Why It's Good For The Game

Its always come off as something that would be very useful to have on-station, and even with just the base assembly components, could be used to great effect in a number of creative situations when you're not trying to burn the entire station down, for the millionth time..

## Changelog
:cl:
add: Adds a new assembly component, the freezer, for gradually making the environment colder.
/:cl:

### CODE REVIEW

File: code/modules/assembly/igniter.dm [hunk 1]
```diff
@@ -42,3 +47,21 @@
...
+
+//For the Condenser, which functions like the igniter but makes things cold.
+/obj/item/assembly/igniter/freezer
+	name = "freezer"
+	desc = "A small electronic device able to chill the surroundings."
+	icon_state = "freezer"
+	custom_materials = list(/datum/material/iron=250, /datum/material/glass=300)
+	heat = 200
+
+/obj/item/assembly/igniter/condenser/activate()
```
Comment (spookydonut) [inline code/modules/assembly/igniter.dm:59]:
```suggestion
/obj/item/assembly/igniter/freezer/activate()
```
i assume

**Comment by MrDoomBringer** [comment]
This and an igniter should be assembly components of the freezer/heater/space heater machines we have

**Comment by interestingusernam3** [comment]
SECURITY TO BAR JANITOR HAS FREEZER

**Comment by Victor239** [comment]
Can we not use the same name that another item already has

**Comment by itseasytosee** [comment]
Dose this cool down nades in assemblies?

**Comment by MrDoomBringer** [comment]
Agreed with @Victor239. Call this a coolant unit or something

**Comment by tmtmtl30** [comment]
igniters have a (highly highly useful) feature (i don't think it's explicitly coded for igniters, since you can do the same with welding tools) where using one on a beaker will heat the beaker up. it'd be neat if the same was present for this component as well, cooling instead of heating.

**Comment by itseasytosee** [comment]
Anything on fire can heat up a beaker and light a cig to my knowledge.

**Comment by tmtmtl30** [comment]
yeah, that was my guess.

**Comment by MrDoomBringer** [comment]
You might be able to do this by just setting the `heat` var to a negative number, rather than implement a whole new system or whatever

**Comment by necromanceranne** [comment]
What happens if I attach an igniter to a freezer?

**Comment by ArcaneMusic** [comment]
> You might be able to do this by just setting the `heat` var to a negative number, rather than implement a whole new system or whatever

I considered that but I had a hunch that it would start creating negative kelvin temperature turfs, and after double checking how hotspots are handled that seems to be right.


> Can we not use the same name that another item already has

Yeah, I think I'll go back to calling these compressors after the thermodynamics component then. I could see this getting confusing.




> What happens if I attach an igniter to a freezer?

Pretty much the same thing that happens when you attach an igniter to an igniter, really.



> igniters have a (highly highly useful) feature (i don't think it's explicitly coded for igniters, since you can do the same with welding tools) where using one on a beaker will heat the beaker up. it'd be neat if the same was present for this component as well, cooling instead of heating.

It currently cools down beakers to just about 200k.

File: code/modules/vending/assist.dm [hunk 2]
```diff
@@ -15,7 +15,8 @@
 					  /obj/item/stock_parts/cell/high = 1)
 	premium = list(/obj/item/price_tagger = 3,
 				   /obj/item/vending_refill/custom = 3,
-				   /obj/item/circuitboard/machine/vendor = 3)
+				   /obj/item/circuitboard/machine/vendor = 3,
+				   /obj/item/assembly/igniter/freezer = 2)
```
Comment (ATH1909) [inline code/modules/vending/assist.dm:19]: can you make these printable from an autolathe?

**Comment by ATH1909** [comment]
Finally, my dreams of a chilled water vapor flamethrower that make icy turfs can become a reality.

... Er, these _do_ work properly in flamethrowers, right?

**Comment by ArcaneMusic** [comment]
> Finally, my dreams of a chilled water vapor flamethrower that make icy turfs can become a reality.
> 
> ... Er, these _do_ work properly in flamethrowers, right?

Flamethrowers specifically don't have a subtype for condensers, but one can be made using a condenser. The actual act for flamethrowers heating tiles is using the actual igniter's _heat_ var now, but it'll probably need a more specialized subtype to do a freezing flame thrower still.

File: code/modules/assembly/igniter.dm [hunk 3]
```diff
@@ -42,3 +47,21 @@
...
+/obj/item/assembly/igniter/condenser
+	name = "condenser"
+	desc = "A small electronic device able to chill their surroundings."
+	icon_state = "freezer"
+	custom_materials = list(/datum/material/iron=250, /datum/material/glass=300)
+	heat = 200
+
+/obj/item/assembly/igniter/condenser/activate()
+	if(!..())
+		return FALSE//Cooldown check
```
Comment (Rohesie) [inline code/modules/assembly/igniter.dm:61]:
```suggestion
	. = ..()
	if(!.)
		return FALSE//Cooldown check
```

File: code/modules/assembly/igniter.dm [hunk 4]
```diff
@@ -42,3 +47,21 @@
...
+	var/turf/location = get_turf(loc)
+	if(location)
+		var/datum/gas_mixture/enviro = location.return_air()
+		enviro.temperature = clamp(min(ROOM_TEMP, enviro.temperature*0.85),MIN_FREEZE_TEMP,MAX_FREEZE_TEMP)
+	sparks.start()
+	return TRUE
```
Comment (Rohesie) [inline code/modules/assembly/igniter.dm:67]:
```suggestion
```

**Review by Twaticus — APPROVED** [review]

File: code/modules/assembly/igniter.dm [hunk 5]
```diff
@@ -1,3 +1,8 @@
+#define EXPOSED_VOLUME 1000
+#define ROOM_TEMP 293
+#define MIN_FREEZE_TEMP 50
+#define MAX_FREEZE_TEMP 1000000
```
Comment (Rohesie) [inline code/modules/assembly/igniter.dm:4]: Undef these at the end of the file if they are not used elsewhere.

And if they are used elsewhere, place them on the defines folder.

**Pull request merged**